File: vuejs/vuejs-tests/src/org/jetbrains/vuejs/language/VueTestSuite.java
// Copyright 2000-2018 JetBrains s.r.o.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
package org.jetbrains.vuejs.language;

import junit.framework.Test;
import junit.framework.TestCase;
import junit.framework.TestSuite;
import org.jetbrains.vuejs.language.parser.VueHighlightingLexerTest;
import org.jetbrains.vuejs.language.parser.VueLexerTest;

public class VueTestSuite extends TestCase {
  public static Test suite() {
    final TestSuite testSuite = new TestSuite(VueTestSuite.class.getSimpleName());
    testSuite.addTestSuite(VueTypedHandlerTest.class);
    testSuite.addTestSuite(VueTypeScriptHighlightingTest.class);
    testSuite.addTestSuite(VueResolveTest.class);
    testSuite.addTestSuite(VueRenameTest.class);
    testSuite.addTestSuite(VueParameterInfoTest.class);
    testSuite.addTestSuite(VueOptimizeImportTest.class);
    testSuite.addTestSuite(VueLiveTemplatesTest.class);
    testSuite.addTestSuite(VueIntentionsTest.class);
    testSuite.addTestSuite(VueInjectionTest.class);
    testSuite.addTestSuite(VueHighlightingTest.class);
    testSuite.addTestSuite(VueFormatterTest.class);
    testSuite.addTestSuite(VueExtractComponentTest.class);
    testSuite.addTestSuite(VueDocumentationTest.class);
    testSuite.addTestSuite(VueCreateTsVariableTest.class);
    testSuite.addTestSuite(VueCopyrightTest.class);
    testSuite.addTestSuite(VueCompletionTest.class);
    testSuite.addTestSuite(VueAutoPopupTest.class);
    testSuite.addTestSuite(VueHighlightingLexerTest.class);
    testSuite.addTestSuite(VueLexerTest.class);
    testSuite.addTestSuite(VueEmmetTest.class);

    return testSuite;
  }
}

File: vuejs/vuejs-tests/src-ext/org/jetbrains/vuejs/lang/VueTestSuite.kt
// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package org.jetbrains.vuejs.lang

import org.jetbrains.vuejs.lang.expr.VueJSParserTest
import org.jetbrains.vuejs.lang.html.VueHighlightingLexerTest
import org.jetbrains.vuejs.lang.html.VueIndexerTest
import org.jetbrains.vuejs.lang.html.VueLexerTest
import org.jetbrains.vuejs.lang.html.VueParserTest
import org.jetbrains.vuejs.libraries.LibrariesTestSuite
import org.jetbrains.vuejs.pug.PugTemplateTest
import org.jetbrains.vuejs.pug.VuePugFoldingTest
import org.jetbrains.vuejs.service.VolarServiceDocumentationTest
import org.jetbrains.vuejs.service.VolarServiceTest
import org.junit.runner.RunWith
import org.junit.runners.Suite

@RunWith(Suite::class)
@Suite.SuiteClasses(
  VueTestCommons::class,
  VueJSParserTest::class,
  VueLexerTest::class,
  VueHighlightingLexerTest::class,
  VueIndexerTest::class,
  VueParserTest::class,
  VueCompletionTest::class,
  VueCompletionTest.WithoutServiceTest::class,
  VueCommenterTest::class,
  VueHighlightingTest::class,
  VueControlFlowTest::class,
  VueComponentTest::class,
  VueNewComponentTest::class,
  VueTypedHandlerTest::class,
  VueAttributeNameParserTest::class,
  VueResolveTest::class,
  VueFindUsagesTest::class,
  VueRenameTest::class,
  VueRenameTest.WithoutServiceTest::class,
  VueCssResolveTest::class,
  VueParameterInfoTest::class,
  VueOptimizeImportTest::class,
  VueLiveTemplatesTest::class,
  VueIntentionsTest::class,
  VueInjectionTest::class,
  VueFormatterTest::class,
  VueRearrangerTest::class,
  VueExternalFilesLinkingTest::class,
  VueExtractComponentTest::class,
  VueDocumentationTest::class,
  VueWebTypesDocumentationTest::class,
  VueCreateTsVariableTest::class,
  VueCopyrightTest::class,
  VueRefAttrsTest::class,
  VueAutoPopupTest::class,
  VueEmmetTest::class,
  VueTypeResolveTest::class,
  VueTemplateInjectionsTest::class,
  LibrariesTestSuite::class,
  PugTemplateTest::class,
  VuePugFoldingTest::class,
  VueModuleImportTest::class,
  VueCopyPasteTest::class,
  VueMoveTest::class,
  VueMoveModuleMemberTest::class,
  VueTypeScriptDuplicateTest::class,
  VueIntroduceVariableTest::class,
  VueTypeScriptLineMarkersTest::class,
  VolarServiceTest::class,
  VolarServiceDocumentationTest::class,
  VueIntegrationHighlightingTest::class,
  VueNpmIntegrationCompletionTest::class,
  VueYarnIntegrationCompletionTest::class,
  // Following tests are not working currently
  //VueESLintHighlightingTest::class,
  //VueTypeScriptWithTslintTest::class,
  //VueTypeScriptHighlightingTest::class,
)
class VueTestSuite
